### ISSUE TITLE
Private repo dependencies no longer being skipped

### ISSUE THREAD

**Issue by philblckwd** [pypa/pip-audit#563]
## Bug description

Private repo dependencies are no longer being skipped if they cannot be resolved.
For example, in the v2.4.14, the audit would succeed and then show that the packages were skipped:
![image](https://user-images.githubusercontent.com/17804114/226451504-ad0d7989-53d1-4d6a-b469-8354cd212153.png)

But now in v2.5.x, the audit fails and shows the error:
![image](https://user-images.githubusercontent.com/17804114/226451919-2f10de89-b57f-4352-98b1-d8f6b6eb843c.png)

## Reproduction steps

- Ensure `pip-audit` v2.5.x is installed
- Create a `requirements.txt` file with a dependency that is installed from a private repo
- Run `pip-audit -r requirements.txt`

## Expected behavior

I'd expect the same behaviour as v2.4.14 - whereby the private dependencies would simply be skipped - unless the `-S` flag was specified.

## Platform information

* OS name and version:
* `pip-audit` version (`pip-audit -V`): 2.5.2
* Python version (`python -V` or `python3 -V`): 3.11.2
* `pip` version (`pip -V` or `pip3 -V`): 23.0.1

**Comment by philblckwd** [comment]
Ah ok gotcha. Yeah this was my first consideration when reviewing the options in the help docs. However, I actually already have the private repo specified in my users `pip.conf` file, but it will seemingly only work when I specify the `extra-index-url` as my private registry, and not the `index-url`.

A.k.a When my `pip.conf` looks like the below, the `pip-audit` fails with the same error I shared above
```
[global]
trusted-host=private.repo.host
index-url=https://private.repo.host/pypi/x/x
extra-index-url=https://pypi.org/simple
```

But when my `pip.conf` looks like the below, the `pip-audit` succeeds, with the expected "skipped" packages, similar to the example screenshot I shared above
```
[global]
trusted-host=private.repo.host
index-url=https://pypi.org/simple
extra-index-url=https://private.repo.host/pypi/x/x
```

Note: I specify the `index-url` as the private repo for security reasons.

**Comment by tetsuo-cpp** [comment]
Yeah, I think the answer is to supply the private indices via the `--extra-index-url` flag. The hashing approach that @woodruffw mentioned works now but we're planning to cleanup this logic to invoke dependency resolution every time so I don't expect this to work in the long run.

> However, I actually already have the private repo specified in my users pip.conf file, but it will seemingly only work when I specify the extra-index-url as my private registry, and not the index-url.

Understood! The issue here is that `pip-audit` doesn't respect `pip.conf`. Supporting `pip.conf` is a feature that we've had in mind for a while now and is captured in #193.

**Comment by philblckwd** [comment]
So I reviewed the code and determined the source of the issue to be the way that `pip-audit` was passing the `index-url` and `extra-index-url` options to the `pip install` cli. `pip-audit` defaults the `index-url` to the PyPI url, therefore it is always passed to the `pip install` CLI - hence why any `index-url` defined in a `pip.conf` file was being ignored. I forked `pip-audit` and created a branch to refactor the `index-url` and `extra-index-url` options so that the `pip install` command would respect the `pip.conf` file unless the `index-url` or `extra-index-url` options were specified. Here's a link to the commit. Let me know what you think and I can open up a PR for it: https://github.com/pypa/pip-audit/commit/1308173f3165be28db685510ab71df42342e5aa5
